numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -95,6 +95,11 @@ def __init__(
                 capacity=config.CACHE_SIZE,
                 pyfunc=self.pyfunc,
             )
+            self._kernel_bundle_cache = LRUCache(
+                name="KernelBundleCache",
+                capacity=config.CACHE_SIZE,
+                pyfunc=self.pyfunc,
+            )
```

Comment:
The _kernel_bundle_cache needs to be defined in the if and else blocks to and set to NullCache.